src/js/components/Director/Director.js

```diff
@@ -0,0 +1,57 @@
+import PropTypes from 'prop-types';
+import React, { useContext, useEffect } from 'react';
+import { withRouter } from 'react-router';
+import { GlobalContext } from '../../contexts/GlobalContextWrapper';
+
+const Director = ({ children, history }) => {
+  const { Config } = useContext(GlobalContext);
+  const { paths: { mainPath, homePath } } = Config.get(['paths']);
+
+  const origin = sessionStorage.getItem('origin');
+  sessionStorage.removeItem('origin');
+
+  const home = ['', mainPath, homePath].join('/');
```

Comment:
This is better than a plain string, but still wouldn't handle duplicate `/`, so next time maybe try to think of a way to join URLs so that they can handle that too